File: algorithms/codeforces/Big Segment.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> l(n), r(n);
    for (int i = 0; i < n; i++) {
        cin >> l[i] >> r[i];
    }

    int min_left = l[0];
    int max_right = r[0];
    for (int i = 1; i < n; i++)  {
        min_left = min(min_left, l[i]);
        max_right = max(max_right, r[i]);
    }

    int res = -1;
    for (int i = 0; i < n; i++) {
        if (l[i] == min_left && r[i] == max_right) {
            res = i + 1;
            break;
        }
    }

    cout << res;
        
    return 0;
}

File: algorithms/python/intermediate/dynamic.array_and_string/Big Segment.py
n = int(input())
min_left = 10 ** 9
max_right = 0
a = []
for _ in range(n):
    l, r = map(int, input().split())
    a.append((l, r))
    min_left = min(min_left, l)
    max_right = max(max_right, r)


res = -1
for i, x in enumerate(a):
    if x[0] == min_left and x[1] == max_right:
        res = i + 1

print(res)